src/args.rs

```diff
@@ -312,6 +312,48 @@ impl Args {
     }
 }
 
+// This macro is used to generate a member function of `ArgMatches` which
+// parses an argument item of the form `[0-9]*(KMG)?`.
+//
+// This is a macro instead of a generic function due to difficulties in
+// correctly constraining the output $ty to work for a number of integers.
+// Since there are limited numbers of types that currently need this function
+// we perform the type monomorphization ourselves.
+macro_rules! human_readable_size_arg_impl {
```

Comment:
You can split this macro into 2 functions: validator and value converter. See [an example of validator](https://github.com/kbknapp/clap-validators/blob/master/src/num/mod.rs)